### PR TITLE
chore(secure_storage): migrate pigeons to use swift and kotlin

### PR DESCRIPTION
**Pull request by Jordan-Nelson**
*Issue #, if available:*

*Description of changes:*
- Update Secure Storage pigeon on Android to use kotlin
- Update NSUserDefaults pigeon on iOS and macOS to use swift

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/secure_storage/amplify_secure_storage/Makefile [hunk 1]
```diff
@@ -1,20 +1,17 @@
...
-	dart format --fix lib/src/messages.cupertino.g.dart
-
-pigeons_macos:
-	dart run pigeon --input pigeons/messages.macos.dart
-	dart format --fix lib/src/messages.cupertino.g.dart
-	# See: https://github.com/flutter/flutter/issues/73738#issuecomment-903927725
-	sed -i '' -e 's/\<Flutter\/Flutter\.h\>/\<FlutterMacOS\/FlutterMacOS\.h\>/' macos/Classes/pigeons/Messages.m
+pigeons_ns_user_defaults:
+	dart run pigeon --input pigeons/ns_user_defaults.dart
+	# TODO(Jordan-Nelson): Consider combining iOS and macOS source code when min
```
Comment (dnys1) [inline packages/secure_storage/amplify_secure_storage/Makefile:13]: Or FFI better yet

File: packages/secure_storage/amplify_secure_storage/pigeons/ns_user_defaults.dart [hunk 2]
```diff
@@ -9,14 +9,15 @@ import 'package:pigeon/pigeon.dart';
...
+    swiftOut: 'ios/Classes/pigeons/NSUserDefaultsPigeon.swift',
     copyrightHeader: 'pigeons/copyright.txt',
   ),
 )
+
+/// A pigeon for interacting with the NSUserDefaults API on iOS and macOS.
 @HostApi()
-abstract class NSUserDefaultsAPI {
+abstract class NSUserDefaultsPigeon {
   @async
```
Comment (dnys1) [inline packages/secure_storage/amplify_secure_storage/pigeons/ns_user_defaults.dart:21]: Can these be sync?

File: packages/secure_storage/amplify_secure_storage/macos/Classes/AmplifySecureStoragePlugin.swift [hunk 3]
```diff
@@ -1,21 +1,21 @@
...
-        NSUserDefaultsAPISetup(registrar.messenger, instance)
+        NSUserDefaultsPigeonSetup.setUp(binaryMessenger: registrar.messenger, api: instance)
     }
     
-    public func setBoolKey(_ key: String, value: NSNumber) async -> FlutterError? {
+    func setBool(key: String, value: Bool, completion: @escaping (Result<Void, Error>) -> Void) {
         UserDefaults.standard.set(value, forKey: key)
-        return nil
+        completion(Result.success(()))
+
```
Comment (dnys1) [inline packages/secure_storage/amplify_secure_storage/macos/Classes/AmplifySecureStoragePlugin.swift:14]:
```suggestion
        completion(.success(()))
```

File: packages/secure_storage/amplify_secure_storage/macos/Classes/AmplifySecureStoragePlugin.swift [hunk 4]
```diff
@@ -1,21 +1,21 @@
...
     }
     
-    public func bool(forKey key: String) async -> (NSNumber?, FlutterError?) {
-        let containsScope = UserDefaults.standard.bool(forKey: key) as NSNumber
-        return (containsScope, nil)
+    func boolFor(key: String, completion: @escaping (Result<Bool, Error>) -> Void) {
+        let value = UserDefaults.standard.bool(forKey: key)
+        completion(Result.success(value))
```
Comment (dnys1) [inline packages/secure_storage/amplify_secure_storage/macos/Classes/AmplifySecureStoragePlugin.swift:19]:
```suggestion
        completion(.success(value))
```

**Review by dnys1 — APPROVED** [review]

**Review by fjnoyp — APPROVED** [review]

File: packages/secure_storage/amplify_secure_storage/ios/Classes/SwiftAmplifySecureStoragePlugin.swift [hunk 5]
```diff
@@ -1,20 +1,21 @@
...
+        NSUserDefaultsPigeonSetup.setUp(binaryMessenger: registrar.messenger(), api: instance)
     }
     
-    public func setBoolKey(_ key: String, value: NSNumber) async -> FlutterError? {
+    func setBool(key: String, value: Bool, completion: @escaping (Result<Void, Error>) -> Void) {
         UserDefaults.standard.set(value, forKey: key)
-        return nil
+        completion(Result.success(()))
+
     }
```
Comment (dnys1) [inline packages/secure_storage/amplify_secure_storage/ios/Classes/SwiftAmplifySecureStoragePlugin.swift:15]: I think this needs to be updated the same way macOS was

**Review by dnys1 — DISMISSED** [review]

**Review by fjnoyp — APPROVED** [review]

**Review by dnys1 — APPROVED** [review]

**Review by fjnoyp — APPROVED** [review]

**Pull request merged**